### PR TITLE
[WIP] Configure Danger

### PR DESCRIPTION
**Pull request by AliSoftware**
First attempt at configuring Danger for this repository

Note: Once this works, we might want to migrate the installation of the `xcpretty` gem into the `Gemfile` as well — now that we have one to include `Danger` — and alter the `.travis.yml` accordingly.

### CODE REVIEW

**Comment by SwiftGen-Eve** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry. <br />
You can find it at <a href="https://github.com/SwiftGen/templates/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
  </tbody>
</table>

Note: we use the following format for CHANGELOG entries:
```
 * [WIP] Configure Danger.  
   [#12](https://github.com/SwiftGen/templates/pull/12)
   [@AliSoftware](https://github.com/AliSoftware)
```
:bulb: Don't forget to use 2 spaces after the full stop at the end of the line describing your changes.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by djbe** [comment]
Need to add `DANGER_GITHUB_API_TOKEN` to the circleci configuration.

**Comment by AliSoftware** [comment]
Done!

btw looking at the `rake compile:output` log files used:
* Their named is only used based on the context being used but not the subcommand. This results in e.g. `output_compile_default_context_empty_raw.log` to be used as a log file for storyboards + strings + colors etc, each overriding the previous one
* it could be worth making it possible to group/organize them in a folder, like `output_compile/strings/default_context_empty_raw.log`

Basically, in `Utils.run`, maybe the `subtask` parameter should add a `_` in the file path but a `/` instead? (even if we might have to `mkdir  -p` the directory first in those cases to ensure it exists)

**Comment by djbe** [comment]
Will look into it. Also saw something reaaaally interesting on Slack today:
https://github.com/orta/danger-junit
Integrates failed tests into danger.
